test/TestProjects/MgmtParamOrdering/Generated/VirtualMachineExtensionImageCollection.cs

```diff
@@ -184,24 +185,23 @@ Page<VirtualMachineExtensionImageResource> FirstPageFunc(int? pageSizeHint)
         /// Operation Id: VirtualMachineExtensionImages_ListVersions
         /// </summary>
         /// <param name="type"> The String to use. </param>
-        /// <param name="filter"> The filter to apply on the operation. </param>
-        /// <param name="top"> The Integer to use. </param>
-        /// <param name="orderby"> The String to use. </param>
+        /// <param name="options"> A class representing the optional parameters in VirtualMachineExtensionImage GetAll method. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentException"> <paramref name="type"/> is an empty string, and was expected to be non-empty. </exception>
         /// <exception cref="ArgumentNullException"> <paramref name="type"/> is null. </exception>
         /// <returns> An async collection of <see cref="VirtualMachineExtensionImageResource" /> that may take multiple service requests to iterate over. </returns>
-        public virtual AsyncPageable<VirtualMachineExtensionImageResource> GetAllAsync(string type, string filter = null, int? top = null, string orderby = null, CancellationToken cancellationToken = default)
+        public virtual AsyncPageable<VirtualMachineExtensionImageResource> GetAllAsync(string type, VirtualMachineExtensionImageGetAllOptions options = null, CancellationToken cancellationToken = default)
```

Comment:
The code needs to leave the old method there if it existed before and follow the guidelines [here](https://github.com/Azure/autorest.csharp/issues/1630#issuecomment-1018265506).